src/adapter/http/src/upload/upload_handler.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use bytes::Bytes;
+use kamu::domain::{ErrorIntoInternal, ResultIntoInternal};
+use serde_json::{json, Value};
+use thiserror::Error;
+
+use crate::api_error::{ApiError, IntoApiError};
+use crate::axum_utils::ensure_authenticated_account;
+use crate::{MakeUploadContextError, SaveUploadError, UploadService};
+
+/////////////////////////////////////////////////////////////////////////////////
+
+#[derive(serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PlatformFileUploadQuery {
+    file_name: String,
+    content_length: usize,
+    content_type: String,
+}
+
+pub async fn platform_file_upload_prepare_post_handler(
+    catalog: axum::extract::Extension<dill::Catalog>,
+    axum::extract::Query(query): axum::extract::Query<PlatformFileUploadQuery>,
+) -> Result<axum::Json<Value>, ApiError> {
+    let account_id =
+        ensure_authenticated_account(&catalog).map_err(ApiError::new_unauthorized_from)?;
```

Comment:
Would be nice to replace `.map_err(ApiError::new_unauthorized_from)?` with just `?` by implementing `IntoApiError` trait.